docs/DEVGUIDE.md

```diff
@@ -45,7 +45,7 @@ To clone the repository:
 First `cd` to the root of the cloned repository tree.
 To build the service-catalog you have two options:
 * `make build`
-* `DOCKER=1 make build`
+* `build/run.sh make build`
```

Comment:
After some offline discussion with @dug, I am personally sold on the approach he is advocating for. Here's what we would tentatively like to do:

- Instead of using `DOCKER=1` to opt-in to a containerized build, we'd like to _default_ to a containerized build. `NO_DOCKER=1` can be used to opt-_out_ of the containerized build.

- Only the containerized build will be officially supported. We know some people do care about the non-containerized build for various reasons, but it's become laborious to test every build process change both in and out-of container. We'd like to stop doing that and let the onus for maintaining the efficacy of non-containerized builds fall on the few who care about it.

- The above means we also propose no longer having Travis build both ways.